app/src/main/java/com/andrew/tasky/agenda/presentation/screens/event_detail/EventDetailFragment.kt

```diff
@@ -63,8 +63,8 @@ class EventDetailFragment : Fragment(R.layout.fragment_event_detail) {
                 viewModel.setEditMode(true)
             }
             header.saveButton.setOnClickListener {
+                progressBar.isVisible = true
```

Comment:
I'd make all your view changes dependent on a state from the ViewModel, otherwise they won't survive config changes like screen rotations. In this case, your `saveEvent()` function from the ViewModel should change an `isSavingEvent` boolean which your fragment observes as a flow or so.
With your approach if you hit save and rotate the device while saving the loading spinner will disappear.